application-forum-ui/src/main/resources/ForumCode/Macros.xml

```diff
@@ -610,15 +610,27 @@
 #end
 ##
 #macro (displayVoteButton $documentReference)
-  ## An user can vote only if didn't already vote and if is authenticated.
-  #set ($canVote = ($services.ratings.getRating($documentReference, $xcontext.userReference) == $util.null) &amp;&amp; !$isGuest)
+  #set ($canVote = false)
+  #if (!$isGuest)
+      ## An user can vote only if didn't already vote and if is authenticated.
+      #set ($pageRating = $services.ratings.getRating($documentReference, $xcontext.userReference))
+      ## On newer versions of services.ratings, the API returns an Optional object, thus we also have to check if the
```

Comment:
Would be good to specify since which version we get an ``Optional`` object so that we can drop the null check.